src/main/java/com/example/javamavenjunithelloworld/HelloApp.java

```diff
@@ -28,7 +28,24 @@ public static void main(String[] args) {
                 System.exit(EXIT_STATUS_PARAMETER_NOT_UNDERSTOOD);
             }
         }
+		int i=0;int j=0;
+					int target = -5;
+		int num = 3;
 
+		target =- num;  // Noncompliant; target = -3. Is that really what's meant?
+		target =+ num; // Noncompliant; target = 3
+		//second bug
+		for (i = 0; i < 10; i++) {
+
+		  i = i - 1; // Noncompliant; counter updated in the body of the loop
+
+		}
+		//third bug
+		for (i = 0; i < 10; j++) {  // Noncompliant
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Move the update of "i" into this loop's update clause. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=squid%3AS1994)